### PR TITLE
Fix GlobalOpenTelemetry usage

### PR DESCRIPTION
**Pull request by heyams**

### CODE REVIEW

**Review by jeanbisutti — APPROVED** [review]

File: sdk/monitor/azure-monitor-opentelemetry-exporter/src/samples/java/com/azure/monitor/opentelemetry/exporter/AzureMonitorMetricExporterSample.java [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+                            });
+
+            Thread.sleep(300000);
+        } catch (Exception ex) {
+            ex.printStackTrace();
+        }
+    }
+
+    // enable proxy via Fiddler
+    private static HttpClient getHttpClientUsingProxy() {
```
Comment (trask) [inline sdk/monitor/azure-monitor-opentelemetry-exporter/src/samples/java/com/azure/monitor/opentelemetry/exporter/AzureMonitorMetricExporterSample.java:122]: I'd probably leave out the proxy configuration

File: sdk/monitor/azure-monitor-opentelemetry-exporter/pom.xml [hunk 2]
```diff
@@ -38,6 +38,7 @@
       --add-opens com.azure.monitor.opentelemetry.exporter/com.azure.monitor.opentelemetry.exporter=ALL-UNNAMED
     </javaModulesSurefireArgLine>
     <checkstyle.excludes>**/implementation/**/*.java</checkstyle.excludes>
+    <opentelemetry.version>1.23.0</opentelemetry.version>
```
Comment (trask) [inline sdk/monitor/azure-monitor-opentelemetry-exporter/pom.xml:41]: looks like this is unused?

File: sdk/monitor/azure-monitor-opentelemetry-exporter/src/main/java/com/azure/monitor/opentelemetry/exporter/AzureMonitorExporterBuilder.java [hunk 3]
```diff
@@ -329,6 +329,7 @@ private HttpPipeline createHttpPipeline() {
         return new HttpPipelineBuilder()
             .policies(policies.toArray(new HttpPipelinePolicy[0]))
             .httpClient(httpClient)
+                .tracer(new NoopTracer())
```
Comment (trask) [inline sdk/monitor/azure-monitor-opentelemetry-exporter/src/main/java/com/azure/monitor/opentelemetry/exporter/AzureMonitorExporterBuilder.java:332]: I think I'd rename this PR to "Fix GlobalOpenTelemetry usage", to make it clearer that it affects customers too

File: sdk/monitor/azure-monitor-opentelemetry-exporter/src/main/java/com/azure/monitor/opentelemetry/exporter/NoopTracer.java [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+    static final AutoCloseable NOOP_CLOSEABLE = () -> {
+    };
+
+    static final Tracer INSTANCE = new NoopTracer();
+
+    NoopTracer() {
+    }
+
+    public Context start(String spanName, Context context) {
```
Comment (trask) [inline sdk/monitor/azure-monitor-opentelemetry-exporter/src/main/java/com/azure/monitor/opentelemetry/exporter/NoopTracer.java:21]: Can you add @Override annotation to all of the interface method implementations?

```suggestion
    @Override
    public Context start(String spanName, Context context) {
```

File: sdk/monitor/azure-monitor-opentelemetry-exporter/src/samples/java/com/azure/monitor/opentelemetry/exporter/AzureMonitorMetricExporterSample.java [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+            OpenTelemetry openTelemetry = OpenTelemetrySdk.builder()
+                    .setMeterProvider(sdkMeterProvider)
+                    .buildAndRegisterGlobal();
+            Meter meter = openTelemetry.meterBuilder("OTEL.AzureMonitor.Demo")
+                    .build();
+            DoubleHistogram histogram = meter.histogramBuilder("histogram").build();
+            histogram.record(1.0);
+            histogram.record(100.0);
+            histogram.record(30.0);
+            Thread.sleep(300000);
```
Comment (trask) [inline sdk/monitor/azure-monitor-opentelemetry-exporter/src/samples/java/com/azure/monitor/opentelemetry/exporter/AzureMonitorMetricExporterSample.java:45]:
```suggestion

            // metrics are exported every 60 seconds by default
            MINUTES.sleep(5);
```

**Review by trask — APPROVED** [review]

**Review by trask — APPROVED** [review]
👍

**Pull request merged**